### PR TITLE
feat: blog redesign

### PR DESCRIPTION
**Pull request by alojzy231**

### CODE REVIEW

File: src/components/ImageWithPlaceholder.tsx [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { FunctionComponent, useState } from 'react';
 import Image, { ImageProps, StaticImageData } from 'next/image';
 
 type ImageWithPlaceholderProps = {
-  placeholderImg: string | StaticImageData;
+  placeholderImg: StaticImageData;
```
Comment (alojzy231) [inline src/components/ImageWithPlaceholder.tsx]: unnecessary removed string

File: src/contextProviders/SearchValueProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { createContext, FunctionComponent, ReactNode, useState, useMemo } from 'react';
```
Comment (alojzy231) [inline src/contextProviders/SearchValueProvider.tsx]: search after clicking on icon or enter

File: src/views/blog/FilteredPosts.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+import GenericPost from './components/GenericPost';
+
+type FilteredPostsProps = {
+  posts: BlogPost[];
+  searchValue: string | undefined;
+};
+
+const FilteredPosts: FunctionComponent<FilteredPostsProps> = ({ posts, searchValue }) => {
+  const [searchPosts, setSearchPosts] = useState<BlogPost[]>([]);
```
Comment (alojzy231) [inline src/views/blog/FilteredPosts.tsx]: useMemo()

File: src/components/SearchInput.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  placeholder,
+  value,
+  onChange,
+  ...restProps
+}) => {
+  const { onToggleMobileNav } = useContext(MobileNavbarContext);
+
+  return (
+    <InputGroup {...restProps}>
+      <Input
```
Comment (alojzy231) [inline src/components/SearchInput.tsx]: variant="outline", style in theme/components

File: src/components/mdx/index.js [hunk 5]
```diff
@@ -15,30 +15,43 @@ import LinkedImage from './LinkedImage';
...
-  p: props => <Text {...props} mt={3} mb={3} />,
-  pre: Pre,
-  code: CodeBlock,
-  strong: Strong,
-  inlineCode: Code,
-  a: Link,
-  ul: List,
-  ol: ListOrdered,
+  h1: (props) => (
+    <Heading as="h2" fontSize={{ base: 'xl', sm: '3xl' }} mb={{ base: 6, sm: 10 }} {...props} />
```
Comment (alojzy231) [inline src/components/mdx/index.js]: change to corresponding heading tag (h1 -> h1)

File: src/consts/placeholderImages.ts [hunk 6]
```diff
@@ -0,0 +1,7 @@
+import PlaceholderOneImage from '@assets/images/placeholders/placeholder1.png';
```
Comment (alojzy231) [inline src/consts/placeholderImages.ts]: direct import

File: src/views/blog/components/GenericPost.tsx [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+  Flex,
+  GridItem,
+  Heading,
+  Text,
+  GridItemProps,
+} from '@chakra-ui/react';
+
+import { BlogPost } from '@typings/BlogPost';
+import ImageWithPlaceholder from '@components/ImageWithPlaceholder';
+import PLACEHOLDERS from '@consts/placeholderImages';
```
Comment (alojzy231) [inline src/views/blog/components/GenericPost.tsx]: direct import

File: src/layouts/MainLayout/MainLayoutHeader.tsx [hunk 8]
```diff
@@ -1,66 +1,60 @@
...
 
-const MainLayoutHeader: FunctionComponent<MainLayoutHeaderProps> = forwardRef(
-  ({ headerCurrentHeight }, ref) => {
-    const { isScrolled } = useScroll(40, true, ref as RefObject<HTMLDivElement>);
-    const { isOpenMobileNav, onToggleMobileNav, isBelowSmallLaptop } = useMainLayoutMobileNav();
-    const headerDesktopPaddingY = isScrolled ? 5 : 8;
+const MainLayoutHeader: FunctionComponent = () => {
+  const ref = useRef(null);
+  const { isScrolled } = useScroll(40, true, ref);
+  const { isBelowSmallLaptop } = useLayoutMobileNav();
```
Comment (alojzy231) [inline src/layouts/MainLayout/MainLayoutHeader.tsx]: useContext()

File: src/pages/blog/[slug].tsx [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+        {CONFIG.app.title} | {frontMatter.metaTitle}
+      </title>
+    </Head>
+    <Box
+      maxW={{ base: '3xl', '1.5xl': 'fit-content' }}
+      mt={{ base: 10, lg: 0 }}
+      mx="auto"
+      px={{ base: 5, '1.5xl': 0 }}
+    >
+      <PostTopSection post={frontMatter as BlogPostData} />
```
Comment (lukasztucholski) [inline src/pages/blog/[slug].tsx]: RC: BlogPostHeader

File: src/pages/blog/[slug].tsx [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+      <PostContent>
```
Comment (lukasztucholski) [inline src/pages/blog/[slug].tsx]: RC: BlogPostContent

File: src/pages/blog/index.tsx [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+
+  return { props: { posts, groupedPosts } };
+}
+
+const BlogPage: NextPage<BlogPageProps> = ({ posts, groupedPosts }) => (
+  <BlogLayout posts={groupedPosts}>
+    <Head>
+      <title>{CONFIG.app.title} | Blog</title>
+    </Head>
+    <TopSection />
```
Comment (lukasztucholski) [inline src/pages/blog/index.tsx]: RC: BlogPageHeader

File: src/pages/blog/index.tsx [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+    <PostsSection posts={posts} />
```
Comment (lukasztucholski) [inline src/pages/blog/index.tsx]: RC: BlogPosts

File: src/views/blog/FeaturedPosts.tsx [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+import TopFeaturedPost from './components/TopFeaturedPost';
+
+type FeaturedPostsProps = { topFeaturedPost: BlogPost; orderedFeaturedPost: BlogPost[] };
+
+const FeaturedPosts: FunctionComponent<FeaturedPostsProps> = ({
+  topFeaturedPost,
+  orderedFeaturedPost,
+}) => (
+  <>
+    <GenericPost post={topFeaturedPost} display={{ base: 'block', '1.5xl': 'none' }} />
```
Comment (lukasztucholski) [inline src/views/blog/FeaturedPosts.tsx]: RC: BlogPostCard

File: src/views/blog/FeaturedPosts.tsx [hunk 14]
```diff
@@ -0,0 +1,24 @@
...
+    <TopFeaturedPost post={topFeaturedPost} display={{ base: 'none', '1.5xl': 'block' }} />
```
Comment (lukasztucholski) [inline src/views/blog/FeaturedPosts.tsx]: RC: FeaturedBlogPost

File: src/views/blog/FeaturedPosts.tsx [hunk 15]
```diff
@@ -0,0 +1,24 @@
...
+
+import { BlogPost } from '@typings/BlogPost';
+
+import GenericPost from './components/GenericPost';
+import TopFeaturedPost from './components/TopFeaturedPost';
+
+type FeaturedPostsProps = { topFeaturedPost: BlogPost; orderedFeaturedPost: BlogPost[] };
+
+const FeaturedPosts: FunctionComponent<FeaturedPostsProps> = ({
+  topFeaturedPost,
```
Comment (lukasztucholski) [inline src/views/blog/FeaturedPosts.tsx]: RC: highlightFeaturedPost or smth like that

File: src/views/blog/FeaturedPosts.tsx [hunk 16]
```diff
@@ -0,0 +1,24 @@
...
+  orderedFeaturedPost,
```
Comment (lukasztucholski) [inline src/views/blog/FeaturedPosts.tsx]: RC: restFeaturedPosts

File: src/views/blog/FeaturedPosts.tsx [hunk 17]
```diff
@@ -0,0 +1,24 @@
+import { FunctionComponent } from 'react';
+
+import { BlogPost } from '@typings/BlogPost';
+
+import GenericPost from './components/GenericPost';
+import TopFeaturedPost from './components/TopFeaturedPost';
+
+type FeaturedPostsProps = { topFeaturedPost: BlogPost; orderedFeaturedPost: BlogPost[] };
+
+const FeaturedPosts: FunctionComponent<FeaturedPostsProps> = ({
```
Comment (lukasztucholski) [inline src/views/blog/FeaturedPosts.tsx]: RC: FeaturedBlogPosts

File: src/views/blog/PostsSection.tsx [hunk 18]
```diff
@@ -0,0 +1,53 @@
...
+const PostsSection: FunctionComponent<PostsSectionProps> = ({ posts }) => {
+  const featuredPosts = posts.filter((post) => post.data.isFeatured);
+  const [topFeaturedPost, ...orderedFeaturedPost] = orderBy(
+    featuredPosts,
+    ['data.featuredOrder', 'data.date'],
+    ['asc', 'asc']
+  );
+
+  const { searchValue } = useContext(SearchValueContext);
+  const deferredSearchValue = useDeferredValue(searchValue);
```
Comment (lukasztucholski) [inline src/views/blog/PostsSection.tsx]: RC: unnecessary deferred value

File: src/views/blog/PostsSection.tsx [hunk 19]
```diff
@@ -0,0 +1,53 @@
...
+import filterPostsBySearchValue from '@utils/filterPostsBySearchValue';
+
+import FeaturedPosts from './FeaturedPosts';
+import SearchPosts from './SearchPosts';
+
+type PostsSectionProps = {
+  posts: BlogPost[];
+};
+
+const PostsSection: FunctionComponent<PostsSectionProps> = ({ posts }) => {
```
Comment (lukasztucholski) [inline src/views/blog/PostsSection.tsx]: RC: BlogPosts

File: src/views/blog/TopSection.tsx [hunk 20]
```diff
@@ -0,0 +1,25 @@
+import { FunctionComponent, useContext, useDeferredValue } from 'react';
+import { Box, Heading, Text } from '@chakra-ui/react';
+
+import { SearchValueContext } from '@contextProviders/SearchValueProvider';
+
+const TopSection: FunctionComponent = () => {
+  const { searchValue } = useContext(SearchValueContext);
+  const deferredSearchValue = useDeferredValue(searchValue);
```
Comment (lukasztucholski) [inline src/views/blog/TopSection.tsx]: RC: unnecessary deferred value

File: src/views/blog/components/CategoryBadge.tsx [hunk 21]
```diff
@@ -0,0 +1,41 @@
...
+const CategoryBadge: FunctionComponent<CategoryBadgeProps> = ({ children, isOnTop, ...props }) => {
+  const { gradients } = useTheme();
+
+  if (isOnTop) {
+    return (
+      <Badge
+        pos="absolute"
+        top="1px"
+        zIndex={1}
+        borderRadius="0 0 4px 0"
```
Comment (lukasztucholski) [inline src/views/blog/components/CategoryBadge.tsx]: TIP: Chakra has props "borderRightBottomRadius" or smth like that.

File: src/views/blog/SlugPost/PostContent.tsx [hunk 22]
```diff
@@ -0,0 +1,21 @@
...
+  children: ReactNode;
+};
+
+const PostContent: FunctionComponent<PostContentProps> = ({ children }) => (
+  <Flex
+    direction="column"
+    maxW="3xl"
+    mx="auto"
+    mt={{ base: 10, '1.5xl': 16 }}
+    fontFamily="'Source Serif Pro', serif"
```
Comment (lukasztucholski) [inline src/views/blog/SlugPost/PostContent.tsx]: Moved it to theme

File: src/views/blog/PostsSection.tsx [hunk 23]
```diff
@@ -0,0 +1,53 @@
...
+  const featuredPosts = posts.filter((post) => post.data.isFeatured);
+  const [topFeaturedPost, ...orderedFeaturedPost] = orderBy(
+    featuredPosts,
+    ['data.featuredOrder', 'data.date'],
+    ['asc', 'asc']
+  );
+
+  const { searchValue } = useContext(SearchValueContext);
+  const deferredSearchValue = useDeferredValue(searchValue);
+  const searchPosts = useMemo(
```
Comment (lukasztucholski) [inline src/views/blog/PostsSection.tsx]: Moved it to Search context

File: src/layouts/BlogLayout/BlogLayoutSidebar/BlogLayoutSidebarSeeAllArticles.tsx [hunk 24]
```diff
@@ -0,0 +1,44 @@
...
+import Link from 'next/link';
+import { AccordionItem, Link as ChakraLink, LinkProps } from '@chakra-ui/react';
+import { ArrowForwardIcon } from '@chakra-ui/icons';
+
+import { SearchValueContext } from '@contextProviders/SearchValueProvider';
+
+const BlogLayoutSidebarSeeAllArticles: FunctionComponent<LinkProps> = ({ ...props }) => {
+  const { handleSearchValueChange } = useContext(SearchValueContext);
+
+  const handleRemoveSearchQuery = (): void => {
```
Comment (lukasztucholski) [inline src/layouts/BlogLayout/BlogLayoutSidebar/BlogLayoutSidebarSeeAllArticles.tsx]: RC: move it to search context

File: src/layouts/BlogLayout/BlogLayoutSidebar/BlogLayoutSidebarSeeAllArticles.tsx [hunk 25]
```diff
@@ -0,0 +1,44 @@
...
+    handleSearchValueChange!('');
+  };
+  return (
+    <AccordionItem w="full" py={3} borderTop="none" onClick={handleRemoveSearchQuery}>
+      <Link href="/blog">
```
Comment (lukasztucholski) [inline src/layouts/BlogLayout/BlogLayoutSidebar/BlogLayoutSidebarSeeAllArticles.tsx]: RC: passHref, move href to const with paths

File: src/layouts/BlogLayout/BlogLayoutSocials.tsx [hunk 26]
```diff
@@ -0,0 +1,29 @@
...
+import { Box, BoxProps, Heading, HStack, Link } from '@chakra-ui/react';
+
+import { LinkedinIcon, TwitterIcon, GitHubIcon } from '@components/icons';
+
+type BlogLayoutSocialsProps = BoxProps;
+
+const BlogLayoutSocials: FunctionComponent<BlogLayoutSocialsProps> = ({ ...props }) => (
+  <Box {...props}>
+    <Heading as="h5" fontSize="sm" fontWeight={500}>
+      FOLLOW OR VISIT US ON
```
Comment (lukasztucholski) [inline src/layouts/BlogLayout/BlogLayoutSocials.tsx]: RC: Let's write in with lowercase and pass props textTransform = uppercase

File: src/layouts/BlogLayout/BlogLayoutSocials.tsx [hunk 27]
```diff
@@ -0,0 +1,29 @@
...
+    </Heading>
+    <HStack gap={5} mt={5}>
+      <Link href="https://www.linkedin.com/company/ockam.io/" isExternal>
```
Comment (lukasztucholski) [inline src/layouts/BlogLayout/BlogLayoutSocials.tsx]: RC: move link to const in global consts folder

File: src/layouts/BlogLayout/BlogLayoutSidebar/BlogLayoutSidebarPostLink.tsx [hunk 28]
```diff
@@ -0,0 +1,37 @@
...
+const BlogLayoutSidebarPostLink: FunctionComponent<BlogLayoutSidebarPostLinkProps> = ({
+  post,
+  ...restProps
+}) => (
+  <ListItem
+    _notLast={{ lg: { mb: 4 } }}
+    bgColor={{ base: 'gray.50', lg: 'inherit' }}
+    {...restProps}
+  >
+    <Link as={`/blog/${post.filePath.replace(/\.mdx?$/, '')}`} href="/blog/[slug]">
```
Comment (lukasztucholski) [inline src/layouts/BlogLayout/BlogLayoutSidebar/BlogLayoutSidebarPostLink.tsx]: RC: passHref & moved 'blog' to paths in consts

File: src/layouts/BlogLayout/BlogLayoutSidebar/BlogLayoutSidebarSeeAllArticles.tsx [hunk 29]
```diff
@@ -0,0 +1,44 @@
...
+        <ChakraLink
+          display="flex"
+          alignItems="center"
+          fontSize="md"
+          fontWeight="regular"
+          color="brand.900"
+          sx={{
+            '&:active svg': {
```
Comment (lukasztucholski) [inline src/layouts/BlogLayout/BlogLayoutSidebar/BlogLayoutSidebarSeeAllArticles.tsx]: _active: {
svg: {
//
}
}

File: src/layouts/BlogLayout/BlogLayoutSocials.tsx [hunk 30]
```diff
@@ -0,0 +1,29 @@
+import { FunctionComponent } from 'react';
+import { Box, BoxProps, Heading, HStack, Link } from '@chakra-ui/react';
+
+import { LinkedinIcon, TwitterIcon, GitHubIcon } from '@components/icons';
+
+type BlogLayoutSocialsProps = BoxProps;
+
+const BlogLayoutSocials: FunctionComponent<BlogLayoutSocialsProps> = ({ ...props }) => (
+  <Box {...props}>
+    <Heading as="h5" fontSize="sm" fontWeight={500}>
```
Comment (lukasztucholski) [inline src/layouts/BlogLayout/BlogLayoutSocials.tsx]: Q: fontWieght="regular" ?

File: src/layouts/BlogLayout/BlogLayoutSocials.tsx (hunk 27 above)
Comment (lukasztucholski) [inline src/layouts/BlogLayout/BlogLayoutSocials.tsx]: RC: render links in loop

**Review by lukasztucholski — CHANGES_REQUESTED** [review]
Let's rename placeholders for blog post to sth like "placeholder-blog-post-silver" etc

**Review by lukasztucholski — APPROVED** [review]
Great work! 🥳

**Pull request merged**